### PR TITLE
feat: Make the FeelEngineBuilder available from Java/Kotlin

### LINKED ISSUE #724: The FeelEngineBuilder is not usable from Java
**Describe the bug**
From a Java application, I can't use the new builder `FeelEngineBuilder`. The (default) constructor is not visible.  

Screenshot from 2023-09-20 13-56-00

**To Reproduce**
1. In a Java class, add the following code snippet:
```
new FeelEngineBuilder().build()
```
2. Verify that the statement doesn't compile 

**Expected behavior**
I can use the new FEEL engine builder from a Java application.

**Environment**
* FEEL engine version: `1.17.0`
* Affects: 
  * Camunda Automation Platform 7: [7.x] 
  * Zeebe broker: [0.x]

### PR DESCRIPTION
**Pull request by vicmosin**
## Description

Refactored implementation

## Related issues

closes #724

### CODE REVIEW

**Comment by vicmosin** [comment]
Mh.. not sure what causes this:

```
[INFO] Comparing to version: 1.17.5
Error:  7002: org.camunda.feel.api.FeelEngineBuilder: Method 'public scala.Function1 curried()' has been removed
Error:  7002: org.camunda.feel.api.FeelEngineBuilder: Method 'public scala.Function1 tupled()' has been removed
```

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala [hunk 1]
```diff
@@ -74,3 +77,9 @@ case class FeelEngineBuilder private (
   )
 
 }
+
+object FeelEngineBuilder {
+
+  @javaStatic
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:83]: :thought_balloon: I don't see that the annotation is required. I can access the static method in the Java class without the annotation.

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala [hunk 2]
```diff
@@ -74,3 +77,9 @@ case class FeelEngineBuilder private (
...
+  def create(): FeelEngineBuilder = FeelEngineBuilder()
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:84]: :+1: Nice.

Comment (saig0) [reply src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:84]: :wrench: Additionally, we could create a convenient method for Java applications. Usually, every Java application will use the JavaValueMapper. 

For example:

```
def forJava(): FeelEngineBuilder = FeelEngineBuilder().withCustomValueMapper(new JavaValueMapper)
```

File: src/test/java/org/camunda/feel/api/FeelEngineBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+package org.camunda.feel.api;
+
+public class FeelEngineBuilderTest {
+
+    public void build() {
+        FeelEngineApi engine = FeelEngineBuilder.create().build();
+    }
+}
```
Comment (saig0) [inline src/test/java/org/camunda/feel/api/FeelEngineBuilderTest.java:24]: :x: We should write a real unit test. 

One option is to use the Java class as a Java application and write a Scala test that uses the Java application.

**Review by saig0 — CHANGES_REQUESTED** [review]
@vicmosin thank you for your contribution. :tada: I added a few comments to improve the solution. Please have a look. :cookie:

**Comment by saig0** [comment]
@vicmosin the build failures are reported by the Clirr plugin. By adding an object to the builder, it changes the Scala internal signature. However, this is okay. We need to exclude these changes in the plugin configuration [here](https://github.com/camunda/feel-scala/blob/main/pom.xml#L254-L349).

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala (hunk 1 above)
Comment (vicmosin) [reply src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala]: Done

File: src/test/java/org/camunda/feel/api/FeelEngineBuilderTest.java (hunk 3 above)
Comment (vicmosin) [reply src/test/java/org/camunda/feel/api/FeelEngineBuilderTest.java]: Not sure if you meant it this way, but I added a real test.

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala (hunk 2 above)
Comment (vicmosin) [reply src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:80]: @saig0 tbh I am not sure I really understood your idea about `forJava` method.. Could you please elaborate?

**Comment by saig0** [comment]
@vicmosin do you want to continue contributing to this PR? The last update was one month ago.

**Comment by vicmosin** [comment]
@saig0 I am sorry, I've been busy last month, will be available at the beginning of the june. If it's urgent, feel free to take over. Thnx

**Comment by saig0** [comment]
@vicmosin thank you for the update. :+1: No need to rush. June is fine. :sun_with_face:

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala (hunk 2 above)
Comment (vicmosin) [reply src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:85]: Done

File: src/test/scala/org/camunda/feel/api/FeelEngineBuilderTest.scala [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+package org.camunda.feel.api
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class FeelEngineBuilderTest extends AnyFlatSpec with Matchers {
+
+  "A FeelEngineBuilder" should "successfully build a FeelEngineApi instance" in {
+    val engine: FeelEngineApi = FeelEngineBuilder.create().build()
```
[the remark this answers is not in the document]
Comment (vicmosin) [reply src/test/scala/org/camunda/feel/api/FeelEngineBuilderTest.scala]: Donr

**Review by saig0 — APPROVED** [review]
I see some minor improvements. However, let's merge this PR first. :rocket:

**Comment by saig0** [comment]
@vicmosin no problem. :laughing: It's merged. :rocket:

**Pull request merged**